Scripts/Editor/Utils/CodeGenerationUtility.cs

```diff
@@ -323,8 +351,9 @@ private static string[] GetCollectionDirectives(ScriptableObjectCollection colle
             indentation++;
             AppendLine(writer, indentation, $"if ({cachedValuesName} == null)");
```

Comment:
With code generation I personally like having template code as a .txt with places in it where you can insert certain parameterization. Something to consider if you're generating a big blob of code, because that gets a bit unreadable sometimes.

Whether that's worth doing here? Up to you. This doesn't look like a massive chunk of code or anything but I did want to mention it.

Reply:
Yeah I might consider looks into it in the future, maybe one of those t4 templates, but yeah not doing that now.